django/db/models/lookups.py

```diff
@@ -171,9 +171,10 @@ def resolve_expression(
         c.lhs = self.lhs.resolve_expression(
             query, allow_joins, reuse, summarize, for_save
         )
-        c.rhs = self.rhs.resolve_expression(
-            query, allow_joins, reuse, summarize, for_save
-        )
+        if hasattr(self.rhs, "resolve_expression"):
```

Comment:
Since `rhs_is_direct_value` tests if `not hasattr(self.rhs, "as_sql")`, we won't prevent the `AttributeError: 'bool' object has no attribute 'resolve_expression` :thinking:

Reply:
Oh I meant the opposite sorry 

```suggestion
        if not self.rhs_is_direct_value():
```

Reply:
Still doesn't seem to work :thinking: 
The tests `lookup.tests.LookupQueryingTests.test_annotate_field_greater_than_field` and `lookup.tests.LookupQueryingTests.test_annotate_literal_greater_than_field` fail because the `rhs` is a `db.models.expressions.F` object.
And `F` defines `resolve_expression` but not `as_sql` :confused:

Reply:
Makes sense, lets keep things as they are then.